goalie/mesh_seq.py

```diff
@@ -531,10 +534,12 @@ def _solve_forward(self, update_solutions=True, solver_kwargs=None):
                     for _ in range(tp.num_timesteps_per_export[i]):
                         next(solver_gen)
                     # Update the solution data
-                    for field, (f, f_) in self.fields.items():
-                        solutions[field].forward[i][j].assign(f)
+                    for field, sol in self.fields.items():
                         if not self.steady:
-                            solutions[field].forward_old[i][j].assign(f_)
+                            solutions[field].forward[i][j].assign(sol[0])
+                            solutions[field].forward_old[i][j].assign(sol[1])
+                        else:
+                            solutions[field].forward[i][j].assign(sol)
```

Comment:
As sanity checks, put in `assert isinstance(sol, tuple)` and `assert isinstance(sol, Function)`, respectively.

Reply:
Added in 6875fd4f80e0e683ad9ece2f9f2ba55a4dc33cd0